### PR TITLE
Add --insecure-skip-tls-verify for repositories

### LINKED ISSUE #5434: Need support for --insecure-skip-tls-verify
Output of `helm version`: 2.13

Output of `kubectl version`: 1.13

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

Need support for `--insecure-skip-tls-verify`

E.g.,

`helm repo update --insecure-skip-tls-verify`

### PR DESCRIPTION
**Pull request by xvzf**
Fixes #5434

### CODE REVIEW

File: pkg/repo/chartrepo.go
Comment (bacongobbler) [inline pkg/repo/chartrepo.go]: is this needed? I don't see a reference to this anywhere.

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Thank you for your contribution. Would you mind writing some tests and documentation to cover this new feature? Thanks.

File: pkg/getter/httpgetter.go
Comment (bacongobbler) [inline pkg/getter/httpgetter.go]: I'm wondering if this should be consolidated with the `if` block above. The `http.ProxyFromEnvironment` and other settings should probably apply here as well , for example (see line 109)

File: pkg/repo/chartrepo.go
Comment (xvzf) [reply pkg/repo/chartrepo.go]: Yes, it seems so - it seems like the struct now including a bool cannot be converted to a string here: https://github.com/xvzf/helm/blob/393fd5fe173540703df320e267b8d706aa9e61b9/cmd/helm/repo_add_test.go#L144

File: pkg/getter/httpgetter.go
Comment (xvzf) [reply pkg/getter/httpgetter.go]: You're right - I'll come up with another way on passing the variable.

Comment (xvzf) [reply pkg/getter/httpgetter.go]: I decided to not consolidate the two if clauses since ca cert check skipping has nothing in common with client auth. Instead I just added `http.ProxyFromEnvironment`.

**Comment by xvzf** [comment]
@bacongobbler Test cases are added and I dropped a line to the docs here: https://github.com/helm/helm-www/pull/440

**Comment by xvzf** [comment]
@bacongobbler Can you have a look? I think the docs for this were already merged (maybe on accident?)

**Comment by xvzf** [comment]
Is there a timeline for this PR? Requested changes are implemented.

**Comment by hmisbah** [comment]
any update on this ?

**Comment by xvzf** [comment]
@hmisbah this is done, awaiting merging.

**Comment by bacongobbler** [comment]
I’ve added this to 3.2 so it’s not lost in the noise. I’m currently away for a few days for an internal meeting. Thanks for your patience.

File: pkg/repo/chartrepo.go
Comment (bacongobbler) [inline pkg/repo/chartrepo.go]: We should probably `panic` here if an error occurs while marshaling. It's very unlikely to occur, but we shouldn't silently ignore the error either.

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Looks good! One more change then should be good to merge.

File: pkg/repo/chartrepo.go
Comment (xvzf) [reply pkg/repo/chartrepo.go]: You are right. In case it fails there it should exit immediately. Fixed that in the latest commit!

**Review by bacongobbler — APPROVED** [review]
LGTM. Will need a second maintainer’s review before merging due to the size of this PR.

**Comment by xvzf** [comment]
👍 Great. I can't add anotherone so let's just wait!

**Review by jdolitsky — APPROVED** [review]

**Comment by TPX01** [comment]
Nice work! I'm testing this actual solution and the adding to repository seems to work fine but the helm pull still seems to ignore this.

```
bamboo@app-bamboo-agent-5:~$ helm pull fluxcd/flux
Error: Get https://<helm-repository>/repository/helm-fluxcd/flux-1.3.0.tgz: x509: certificate signed by unknown authority
```

```
bamboo@app-bamboo-agent-5:~$ helm repo list
NAME    URL
fluxcd  https://<helm-repository>/repository/helm-fluxcd/
```

Was this insecure feature only meant for adding a repo to helm?

**Comment by xvzf** [comment]
Thanks for the feedback @TPX01
I'll have a look at it.

Which helm version are you using?

**Comment by xvzf** [comment]
I was able to reproduce it; can you open an issue?

I'll address it asap!

**Comment by TPX01** [comment]
> I was able to reproduce it; can you open an issue?
> 
> I'll address it asap!

Thanks alot @xvzf ! I've created the ticket: https://github.com/helm/helm/issues/7875

Le me know if that is enough information to get this going. I'm rather new to the actual code but it looks like the downloader is always using TLS veritification.

**Comment by xvzf** [comment]
Should be sufficient, I’ll have to figure out where it fails since the getters are configured to respect the option..

**Pull request merged**
